### PR TITLE
inverse matching of tags. Issue #10186

### PR DESCRIPTION
**Pull request by vktg**
- [ ] Redmine Issue: https://redmine.pfsense.org/issues/10186
- [ ] Ready for review

allow to inverse tagged rules
i.e.
`block out quick on RED_NET ! tagged RED_NET_OK`

### CODE REVIEW

File: src/usr/local/www/firewall_rules_edit.php [hunk 1]
```diff
@@ -521,6 +525,10 @@ function is_aoadv_used($rule_config) {
 		$input_errors[] = gettext("Invert match cannot be selected with 'any'.");
 	}
 
+	if ((isset($_POST['nottagged']) && empty($_POST['tagged']))) {
```
Comment (jim-p) [inline src/usr/local/www/firewall_rules_edit.php]: There is an extra set of parenthesis on the outside this test

File: src/usr/local/www/firewall_rules_edit.php [hunk 2]
```diff
@@ -1570,12 +1583,25 @@ function build_flag_table() {
...
-	'Tagged',
-	'text',
-	$pconfig['tagged']
-))->setHelp('A packet can be matched on a mark placed before on another rule.');
+$group = new Form_Group('Tagged');
+
+$group->add(new Form_Checkbox(
+	'nottagged',
+	'nottagged',
+	'Invert',
```
Comment (jim-p) [inline src/usr/local/www/firewall_rules_edit.php]: This should be `Invert match`

File: src/usr/local/www/firewall_rules_edit.php [hunk 3]
```diff
@@ -1570,12 +1583,25 @@ function build_flag_table() {
...
+))->setWidth(1);
+
+$group->add(new Form_Input(
+'tagged',
+'Tagged',
+'text',
+$pconfig['tagged']
+))->setWidth(4);
+
+$group->setHelp('A packet can be matched on a mark placed before on another rule.');
```
Comment (jim-p) [inline src/usr/local/www/firewall_rules_edit.php]: Change this to:
> Match a mark placed on a packet by a different rule with the Tag option. Check Invert match to match packets which do not contain this tag.

File: src/usr/local/www/firewall_rules_edit.php (hunk 1 above)
Comment (vktg) [reply src/usr/local/www/firewall_rules_edit.php]: fixed

File: src/usr/local/www/firewall_rules_edit.php (hunk 3 above)
Comment (vktg) [reply src/usr/local/www/firewall_rules_edit.php]: done

File: src/usr/local/www/firewall_rules_edit.php (hunk 2 above)
Comment (vktg) [reply src/usr/local/www/firewall_rules_edit.php]: done,
but now GUI is not so pretty

Comment (jim-p) [reply src/usr/local/www/firewall_rules_edit.php]: I see that now, it forces a word wrap. I guess it's OK to move it back to just 'Invert' and adjust the text below to match.

Comment (vktg) [reply src/usr/local/www/firewall_rules_edit.php]: done

**Pull request merged**